modules/user/user.controllers.js

```diff
@@ -23,6 +28,29 @@ const mixins = {
     }
     return data;
   },
+
+  async loginUsingWallet(signature, signPayload, { clientIpAddress }) {
+    const { address } = WalletUtils.validateSignature(signature, signPayload, {
+      ip: clientIpAddress,
+      secret,
+    });
+    let user = await this.table.findOne({
+      where: {
+        wallet_address: address,
+      },
+    });
+    let data = await this.loginSuccess(user.id, clientIpAddress);
+    if (data.user.roles?.includes('donor')) {
+      let keys = getPrivateKeys('donor');
+
+      data.privateKey = keys.privateKey;
+    } else {
+      let keys = getPrivateKeys('admin');
```

Comment:
No need to send private key if login from wallet. user use their metamask to do transaction.